pkg/kubelet/kubelet.go

```diff
@@ -2457,6 +2457,61 @@ func (kl *Kubelet) setNodeStatus(node *api.Node) error {
 			kl.recordNodeStatusEvent("NodeNotReady")
 		}
 	}
+
```

Comment:
I definitely agree. The pattern and logic here is hard to follow. I spent my entire first week just trying to understand the pattern and logic here. Also, the fact that this file and its accompanying test are gigantic aggravates things.

Added a TODO.